### ISSUE TITLE
Potential performance issue since 3.9

### ISSUE THREAD

**Issue by mboynes** [alleyinteractive/wordpress-fieldmanager#135]
It looks like 3.9 introduced some potentially inefficient code in `wp_enqueue_media()`. It now counts all attachments on the site as follows:

```
SELECT post_mime_type, COUNT( * ) AS num_posts FROM wp_posts WHERE post_type = 'attachment' AND post_status != 'trash' GROUP BY post_mime_type
include('wp-admin/admin-header.php'), do_action('admin_print_scripts'), call_user_func_array, Fieldmanager_Media->{closure}, wp_enqueue_media, wp_count_attachments #118 (1,174.6ms)
```

I noticed on a site with 50k rows in `wp_posts` this query was taking over a second to run. While this is mostly an issue with core (which I'll file a trac ticket on), it's only an issue on editing pages. The problem in FM is that FM always enqueues media if there's a media field defined.

For reference, this code was tracked through ticket [27554](https://core.trac.wordpress.org/ticket/27554), and was added in [this commit](https://github.com/WordPress/WordPress/commit/ff6c0731fa8f04b4614220c89f1c41be6ddfa2b3#diff-297bf46a572d5f80513d3fed476cd2a2).

**TL;DR: Any instance of Fieldmanager_Media will call `wp_enqueue_media()` on every admin page, but this function may slow things down significantly. FM should only call `wp_enqueue_media()` on pages where it has media fields.**
